__tests__/integrations/BingAds/browser.test.js

```diff
@@ -0,0 +1,148 @@
+import { BingAds } from '../../../src/integrations/BingAds';
+import { event, query, products } from './__mocks__/data';
+
+describe('BingAds init tests', () => {
+  let Bingads;
+  test('Testing init call of Bing ads', () => {
+    Bingads = new BingAds({ tagID: '12567839' }, { loglevel: 'debug' });
+    Bingads.init();
+    expect(typeof window.bing12567839).toBe('object');
+  });
+});
+
+describe('BingAds page', () => {
+  let bingAds;
+  const output = [];
+  beforeEach(() => {
+    bingAds = new BingAds({ tagID: '12567839' }, { loglevel: 'debug' });
+    bingAds.init();
+    window.bing12567839.push = jest.fn((x) => output.push(x));
+  });
+
+  test('send pageview', () => {
+    bingAds.page({
+      context: {},
+      properties: {
+        category: 'test category',
+        path: '/test',
+        url: 'http://localhost',
+        referrer: '',
+        title: 'test page',
+        testDimension: 'abc',
+      },
+    });
+    expect(output[0]).toEqual('pageLoad');
+  });
+});
+
+describe('BingAds Track event', () => {
+  let bingAds;
+  const output = [];
+  beforeEach(() => {
+    bingAds = new BingAds({ tagID: '12567839' }, { loglevel: 'DEBUG' });
+    bingAds.init();
+    window.bing12567839.push = jest.fn((x, y, z) => output.push({ event: y, ...z }));
+  });
+
+  test('Test for all properties not null', () => {
+    bingAds.track({
+      message: {
+        type: 'track',
+        context: {},
+        event,
+        properties: {
+          event_action: 'button_click',
+          category: 'Food',
+          currency: 'INR',
+          total: 18.9,
+          value: 20,
+          revenue: 25.5,
+          ecomm_category: '80',
+          transaction_id: 'txn-123',
+          ecomm_pagetype: 'Cart',
+          query,
+          products,
+        },
+      },
+    });
+    expect(output[0]).toEqual({
+      event: 'button_click',
+      event_label: event,
+      event_category: 'Food',
+      currency: 'INR',
+      revenue_value: 18.9,
+      search_term: query,
+      ecomm_query: query,
+      ecomm_category: '80',
+      transaction_id: 'txn-123',
+      ecomm_pagetype: 'Cart',
+      ecomm_prodid: ['123', '345'],
+      items: [
+        { id: '123', price: 14.99, quantity: 2 },
+        { id: '345', price: 3.99, quantity: 1 },
+      ],
+      ecomm_totalvalue: 18.9,
+    });
+  });
+
+  test('Test for custom properties', () => {
+    bingAds.track({
+      message: {
+        type: 'track',
+        context: {},
+        event,
+        properties: {
+          event_action: 'button_click',
+          category: 'Food',
+          currency: 'INR',
+          customProp: 'custom',
+        },
+      },
+    });
+    expect(output[1]).toEqual({
+      event: 'button_click',
+      event_label: event,
+      event_category: 'Food',
+      currency: 'INR',
+      customProp: 'custom',
+      ecomm_pagetype: 'other',
```

Comment:
From where we are generating this `event` field?

Reply:
properties.event_action or message.type